subsys/bluetooth/controller/ll_sw/nordic/lll/lll_scan.c

```diff
@@ -1549,6 +1549,11 @@ static int isr_rx_scan_report(struct lll_scan *lll, uint8_t devmatch_ok,
 			{
 				struct node_rx_ftr *ftr;
 
+				/* Reset Scan context association with any Aux context as a new
+				 * extended advertising chain is being setup for reception here.
+				 */
+				lll->lll_aux = NULL;
```

Comment:
if `lll_aux` is non-NULL before this, we don't need to do any cleanups or similar?

Reply:
No, this is just an association do that received PDUs are added to a linked list in the aux context. By setting it to NULL, a new aux context will get associated to which the new chain of PDUs will be added to the linked list.

Reply:
Found and issue in overnight testing :-(
This code has a race and is causing the LLL scheduled auxiliary PDU reception to leak aux contexts :-(

Reply:
Fixed now... answers to the questions I had asked myself, avoid both set and reset from ULL execution context. Now association reset in LLL, and association set in ULL, this solves aux context memory leak.